tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
+ * @author Jessica Quynh Tran
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  create(context) {
+
+    const msg = 'Use of `let` declaration within a for-loop is ' +
+                'not recommended. Please use `var` instead.';
+
+    /**
+     * Report function to test if the for-loop is declared using `let`.
+     */
+    function testForLoop(node) {
+      if (node.init && node.init.kind === 'let') {
+        context.report(node, msg);
+      }
+    }
+
+    return {
+      'ForStatement': (node) => testForLoop(node),
+      'ForInStatement': (node) => testForLoop(node),
+      'ForOfStatement': (node) => testForLoop(node)
```

Comment:
The implementation for `ForStatement` nodes looks good. I think checking `ForInStatement` and `ForOfStatement` nodes is a good idea, but I don't think this rule is correctly checking those nodes at the moment, since they don't have an `init` property. Instead, I think you would want to check the `left` property.